ShaiRandom/Generators/KnownSeriesRandom.cs

```diff
@@ -656,22 +649,44 @@ public ulong SelectState(int selection)
         ///     - 5: floatSeries
         ///     - 6: longSeries
         ///     - 7: ulongSeries
+        ///     - 8: decimalSeries
         /// </remarks>
         /// <param name="selection">Selection value of index to set.</param>
         /// <param name="value">Value to set the index to.</param>
         public void SetSelectedState(int selection, ulong value)
         {
+            int state = (int)value;
             switch (selection)
             {
-                case 0: _intIndex = (int)value; break;
-                case 1: _uintIndex = (int)value; break;
-                case 2: _doubleIndex = (int)value; break;
-                case 3: _boolIndex = (int)value; break;
-                case 4: _byteIndex = (int)value; break;
-                case 5: _floatIndex = (int)value; break;
-                case 6: _longIndex = (int)value; break;
-                case 8: _decimalIndex = (int)value; break;
-                default: _ulongIndex = (int)value; break;
+                case 0:
+                    _intIndex = _intSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _intSeries.Count);
+                    break;
+                case 1:
+                    _uintIndex = _uintSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _uintSeries.Count);
+                    break;
+                case 2:
+                    _doubleIndex = _doubleSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _doubleSeries.Count);
+                    break;
+                case 3:
+                    _boolIndex = _boolSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _boolSeries.Count);
+                    break;
+                case 4:
+                    _byteIndex = _byteSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _byteSeries.Count);
+                    break;
+                case 5:
+                    _floatIndex = _floatSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _floatSeries.Count);
+                    break;
+                case 6:
+                    _longIndex = _longSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _longSeries.Count);
+                    break;
+                case 7:
+                    _ulongIndex = _ulongSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _ulongSeries.Count);
+                    break;
+                case 8:
+                    _decimalIndex = _decimalSeries.Count == 0 ? 0 : MathUtils.WrapAround(state, _decimalSeries.Count);
+                    break;
```

Comment:
I think the reason I didn't check the indices here was because in other RNG classes, I had used properties for states that had invalid possible settings, and had the setters check. It's also fine to have the check here.